assignment/scripts/3-music-collection.js

```diff
@@ -1 +1,144 @@
-console.log('***** Music Collection *****')
+console.log('***** Music Collection *****');
+
+// creating empty collection
+let collection = [];
+
+// start addToCollection 
+function addToCollection(title, artist, yearPublished, tracks) {
+
+    // create album object
+    var album = {
+        title: title,
+        artist: artist,
+        yearPublished: yearPublished,
+        trackList: tracks
+    };
+
+    // push the album to the collection & return the album
+    collection.push(album);
+    return album;
+}
+// end addToCollection
+
+// adding albums by testing addToCollection
+console.log('Adding album: ', addToCollection('Ride the Lightning', 'Metallica', 1984, [{name: 'Ride the Lightning', duration: '6:36'}, {name: 'For Whom The Bell Tolls', duration: '5:09'}]));
+console.log('Adding album: ', addToCollection('III', 'Lumineers', 2019, [{name: 'Gloria', duration: '3:36'}]));
+console.log('Adding album: ', addToCollection('717 Tapes', 'Warren Zeiders', 2022, [{name: 'Up To No Good', duration: '2:13'}]));
+console.log('Adding album: ', addToCollection('Garage, Inc.', 'Metallica', 1998, [{name: 'Whiskey in the Jar', duration: '5:04'}]));
+console.log('Adding album: ', addToCollection('Dance Fever', 'Florence and the Machine', 2022, [{name: 'Free', duration: '3:54'}]));
+console.log('Adding album: ', addToCollection('Orange Blood', 'Mt. Joy', 2022, [{name: 'Bathroom Light', duration: '3:08'}]));
+console.log('The full collection is: ', collection);
+
+console.log('\n -------------- Ending addToCollection, starting showCollection --------------\n ');
+
+// start showCollection
+function showCollection(array) {
+
+    // logging the number of albums
+    console.log(`The collection is ${array.length} albums.`);
+
+    // looping through each album, logging values with . operator
+    for (let album of array) {
+        i = 1;
```

Comment:
Don't forget the `let`!

JavaScript unfortunately won't yell at you if you forget to declare your variables (unless you are in strict mode), but doing so can trigger weird behavior that is difficult to debug. You end up creating a "global property" instead of a variable. It exists everywhere and can conflict with other things named `i`.